conman/routes/models.py

```diff
@@ -74,6 +84,64 @@ def handle(self, request, path):
         # Deal with the request
         return handler.handle(request, path)
 
+    def move_to(self, new_url, *, move_children):
+        """
+        Move this Route to a new url.
+
+        Requires `move_children` as a keyword argument. If True, all child
+        Routes will be moved along with this parent. Otherwise, when False, any
+        and all children will remain where they are.
+
+        In the event of a clash, an IntegrityError will be raised.
+        """
+        if move_children:
+            Route.objects.move_branch(self.url, new_url)
+            self.url = new_url
+        else:
+            old_url, self.url = self.url, new_url
+            try:
+                self.save()
+            except IntegrityError:
+                self.url = old_url
+                raise
+
+    def swap_with(self, other_route, *, move_children):
+        """
+        Swap this Route's URL with that of another Route.
+
+        Requires `move_children` as a keyword argument. When `True`, all
+        descendant routes will be moved along with the parents. When `False`,
+        they will remain where they are, and only the parents will be swapped.
+
+        When one of the `Routes` is a descendant of the other, `move_children`
+        must be `False`.
+
+        If DEFERRED unique contrstraints get supported in django, we might
+        benefit from revisiting how this works (ie: dropping the UUID).
+
+        See https://code.djangoproject.com/ticket/20581.
+        """
+        if not (self.pk and other_route.pk):
+            raise ValueError('Cannot move unsaved Routes.')
+
+        urls = sorted((self.url, other_route.url))
+        if move_children and urls[1].startswith(urls[0]):
+            msg = 'Cannot move children when swapping ancestors with descendants.'
+            raise ValueError(msg)
+
+        tmp_path = str(uuid.uuid4())
+        if move_children:
+            Route.objects.move_branch(self.url, tmp_path)
+            Route.objects.move_branch(other_route.url, self.url)
+            Route.objects.move_branch(tmp_path, other_route.url)
+            other_route.url, self.url = self.url, other_route.url
```

Comment:
`move_branch` will update these rows in the database, but this gives us a chance to update the values of these objects without a call to `refresh_from_db` (which would be required, as `move_branch` uses `Route.objects.update()`).